src/adapter/graphql/src/scalars/flow_configuration.rs

```diff
@@ -91,8 +91,13 @@ pub enum TimeUnit {
 
 impl From<chrono::Duration> for TimeDelta {
     fn from(value: chrono::Duration) -> Self {
+        if value.num_seconds() <= 0 {
+            panic!("Positive interval expected, but received [{value}]")
+        }
+
         let num_weeks = value.num_weeks();
         if (value - chrono::Duration::weeks(num_weeks)).is_zero() {
+            #[allow(clippy::cast_sign_loss, clippy::cast_possible_truncation)]
             return Self {
                 every: num_weeks as u32,
                 unit: TimeUnit::Weeks,
```

Comment:
In this case, I think it's fine to convert `every` to i64.

Reply:
Done